app/src/test/java/org/simple/clinic/sync/indicator/SyncIndicatorViewControllerTest.kt

```diff
@@ -149,4 +178,18 @@ class SyncIndicatorViewControllerTest {
     verify(dataSync, never()).streamSyncErrors()
     verify(indicator, never()).showErrorDialog(any())
   }
+
+  @Test
+  fun `if any record is pending sync, then sync indicator should show Sync Pending`() {
+    whenever(patientRepository.pendingRecordsCount()).thenReturn(Observable.just(1))
+    whenever(bpRepository.pendingRecordsCount()).thenReturn(Observable.just(1))
+    whenever(prescriptionRepository.pendingRecordsCount()).thenReturn(Observable.just(1))
+    whenever(medicalHistoryRepository.pendingRecordsCount()).thenReturn(Observable.just(1))
+    whenever(appointmentRepository.pendingRecordsCount()).thenReturn(Observable.just(0))
+    whenever(communicationRepository.pendingRecordsCount()).thenReturn(Observable.just(0))
```

Comment:
I think we should test this for each data type separately instead of clubbing them together in one test. @vinaysshenoy what do you think?

Reply:
Added test for each record.